lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1006,10 +1006,10 @@ struct LoadOpConversion
       assert(wordNElems * nWords * numVecs == numElems);
 
 #ifdef USE_ROCM
-      // TODO: need to revisit this when implementing vectorized loads
-      Value ret = load(ptrElems[vecStart]);
-      ret = bitcast(ret, valueElemTy);
-      loadedVals.push_back(ret);
+      for (size_t wordIdx = 0; wordIdx < nWords; ++wordIdx) {
+        Value ret = load(ptrElems[vecStart + wordIdx]);
+        loadedVals.push_back(ret);
```

Comment:
i think we need a bitcase here:

ret = bitcast(ret, valueElemTy);

Reply:
Done